### PR TITLE
Create Report Definition Final UI Changes 

### PR DESCRIPTION
**Pull request by davidcui-amzn**
*Issue #, if available:*
N/A
*Description of changes:*
Address UI-strict changes in `Create report definition` as discussed in UI/UX Final review
* Added hyphen to `Cron-based` label
* Added external link to Opendistro cron docs for `Cron help` link (TODO: link to reporting cron vs alerting cron doc) 
* Fixed email delivery section so text doesn't disappear on option change
* Added failure toast when unsuccessful report definition creation is attempted

TODO:
* Add success and share toasts (pending functionality addition) 
* Connect `Edit report definition` correctly
* Address attachment checkbox in email (will be added to this PR later today after sync with UX) 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: kibana-reports/public/components/report_definitions/report_trigger/report_trigger_constants.tsx [hunk 1]
```diff
@@ -31,7 +31,7 @@ export const SCHEDULE_REQUEST_TIME_OPTIONS = [
   },
   {
     id: 'cronBasedOption',
-    label: 'Cron based',
+    label: 'Cron-based',
```
Comment (zhongnansu) [inline kibana-reports/public/components/report_definitions/report_trigger/report_trigger_constants.tsx:34]: why is this changed? I assume this label field will be the value that client passes to server, which is defined as `Cron based` in schema

File: kibana-reports/public/components/report_definitions/report_trigger/report_trigger.tsx [hunk 2]
```diff
@@ -353,7 +353,7 @@ export function ReportTrigger(props) {
           label="Custom cron expression"
           labelAppend={
             <EuiText size="xs">
-              <EuiLink href="#">Cron help</EuiLink>
+              <EuiLink href="https://opendistro.github.io/for-elasticsearch-docs/docs/alerting/cron/">Cron help</EuiLink>
```
Comment (zhongnansu) [inline kibana-reports/public/components/report_definitions/report_trigger/report_trigger.tsx:356]: this is great!

**Review by akbhatta — APPROVED** [review]

File: kibana-reports/public/components/report_definitions/delivery/delivery.tsx [hunk 3]
```diff
@@ -40,6 +40,8 @@ import {
   EMAIL_RECIPIENT_OPTIONS,
 } from './delivery_constants';
 import 'react-mde/lib/styles/css/react-mde-all.css';
+// import 'babel-loader';
+// import './style/css/react-mde-custom-style.css';
```
Comment (zhongnansu) [inline kibana-reports/public/components/report_definitions/delivery/delivery.tsx:44]: can these be removed?

File: kibana-reports/public/components/report_definitions/report_trigger/report_trigger_constants.tsx (hunk 1 above)
Comment (davidcui-amzn) [reply kibana-reports/public/components/report_definitions/report_trigger/report_trigger_constants.tsx:34]: This is only the label that the user will see, and a hyphen was requested in the UI final review.

File: kibana-reports/public/components/report_definitions/delivery/delivery.tsx (hunk 3 above)
Comment (davidcui-amzn) [reply kibana-reports/public/components/report_definitions/delivery/delivery.tsx:44]: Ack- good catch, will remove

**Review by zhongnansu — APPROVED** [review]
